src/sentry/static/sentry/app/views/projectGeneralSettings.jsx

```diff
@@ -134,17 +190,63 @@ export default class ProjectGeneralSettings extends AsyncView {
       );
     } else {
       return (
-        <p>
-          <a
-            href={`/${orgId}/${projectId}/settings/transfer/`}
-            className="btn btn-danger pull-right"
+        <Field
+          label={tct('Transfer the [project] project and all related data.', {
+            project: <strong>{project.slug}</strong>,
+          })}
+          help={t('Careful, this action cannot be undone.')}
+        >
+          <Confirm
+            onConfirm={this.handleTransferProject}
+            priority="danger"
+            title={`${t('Transfer project')}?`}
+            confirmText={t('Transfer project')}
+            renderMessage={({doConfirm, doClose}) => (
+              <div>
+                <TextBlock>
+                  <strong>
+                    {t('Transferring this project is permanent and cannot be undone!')}
+                  </strong>
+                </TextBlock>
+                <TextBlock>
+                  {t(
+                    'Please enter the owner of the organization you would like to transfer this project to.'
+                  )}
+                </TextBlock>
+                <Panel>
+                  <PanelHeader>{t('Transfer to')}</PanelHeader>
+                  <Form
+                    hideFooter
+                    onFieldChange={this.handleTransferFieldChange}
+                    onSubmit={(data, onSuccess, onError, e) => {
+                      e.stopPropagation();
+                      doConfirm();
+                    }}
+                  >
+                    <NewTextField
```

Comment:
Would it be possible to just use the new one throughout this file? So we don't have to call it new text field :)

Reply:
This is taken care of in #7311